src/App.jsx

```diff
@@ -14,57 +16,71 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
+  const clearSelection = () => setSelectedGood('');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  return (
+    <main className="section container">
+      <h1
+        className={classNames('title', {
+          'is-flex is-align-items-center': selectedGood,
+        })}
+      >
+        {selectedGood ? `${selectedGood} is selected` : 'No goods selected'}
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+        {selectedGood && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={clearSelection}
+          />
+        )}
+      </h1>
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+      <table className="table">
+        <tbody>
+          {goods.map(good => {
+            const isSelected = selectedGood === good;
+            const handleSelection = isSelected
+              ? clearSelection
+              : () => setSelectedGood(good);
```

Comment:
define functions like handleSelection outside of the JSX. This ensures that the logic remains consistent across renders, avoids unnecessary re-creation of the function, and optimizes performance. Additionally, keeping the logic separated from the JSX makes the code cleaner, more maintainable, and easier to test.